### PR TITLE
🌱 improve logging during Machine deletion

### PR DESCRIPTION
**Pull request by schrej**
**What this PR does / why we need it**:
Fixes duplicated log output (lines 371 and 385) introduced with #6413 and adds two more outputs when waiting for infrastructure or bootstrap deletion.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

### CODE REVIEW

File: internal/controllers/machine/machine_controller.go [hunk 1]
```diff
@@ -412,10 +410,16 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 	}
 
 	if ok, err := r.reconcileDeleteInfrastructure(ctx, m); !ok || err != nil {
+		if err == nil {
+			log.Info("Waiting for infrastructure to be deleted", "Infrastructure", klog.KRef("", m.Spec.InfrastructureRef.Name))
+		}
 		return ctrl.Result{}, err
 	}
```
Comment (sbueringer) [inline internal/controllers/machine/machine_controller.go:417]:
```suggestion
infrastructureDeleted, err := r.reconcileDeleteInfrastructure(ctx, m)
	if err != nil {
		return ctrl.Result{}, err
	}
	if !infrastructureDeleted {
		log.Info("Waiting for infrastructure to be deleted", m.Spec.InfrastructureRef.Kind, klog.KRef(m.Spec.InfrastructureRef.Namespace, m.Spec.InfrastructureRef.Name))
		return ctrl.Result{}, nil
	}
```

WDYT about something like this instead? Just to make it very clear and readable.

Comment (sbueringer) [reply internal/controllers/machine/machine_controller.go]: @schrej Can we also use the actual Kind as key + the Namespace in klog.KRef?

Comment (schrej) [reply internal/controllers/machine/machine_controller.go]: Ah, sorry, missed that. Makes sense, was copy pasting from that log line related to Node, but that's different of course.

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by sbueringer** [comment]
Thx!
/approve

**Pull request merged**